source/Octopus.Tentacle/Communications/AutofacServiceFactory.cs

```diff
@@ -49,8 +37,12 @@ public IServiceLease CreateService(string serviceName)
             {
                 if (knownServices.TryGetValue(serviceName, out var knownService))
                 {
-                    //because the service implementations are registered `AsSelf()`, we can resolve them directly
-                    return new Lease(scope.Resolve(knownService.ServiceImplementationType));
+                    //create a nested scope for the service lease. Halibut will automatically dispose of this lease at the end of the RPC call,
+                    //so pass the nested scope to the lease for disposal
+                    var nestedScope = scope.BeginLifetimeScope();
```

Comment:
Just wondering, what was the motivation behind creating a nested scope here instead of just resolving the service from `scope`?

Reply:
See #860